### PR TITLE
HIPP-1444: Migration to hard delete applications without access requests

### PR DESCRIPTION
**Pull request by PaulCDurham**
https://jira.tools.tax.service.gov.uk/browse/HIPP-1444

### CODE REVIEW

File: app/uk/gov/hmrc/apihubapplications/mongojobs/HardDeleteMigrationJob.scala [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    def add(result: MigrationResult): MigrationSummary = {
+      result match {
+        case NotDeleted => this.copy(applications = applications + 1)
+        case SoftDeleted => this.copy(applications = applications + 1, softDeleted = softDeleted + 1)
+        case HardDeleted => this.copy(applications = applications + 1, hardDeleted = hardDeleted + 1)
+        case Failed => this.copy(applications = applications + 1, failed = failed + 1)
+      }
+    }
+
+    def deleted: Int = this.softDeleted + this.hardDeleted + this.failed
```
Comment (victorarbuesmallada) [inline app/uk/gov/hmrc/apihubapplications/mongojobs/HardDeleteMigrationJob.scala:110]: should the failed deletions count in this total?

File: app/uk/gov/hmrc/apihubapplications/mongojobs/MongoJobModule.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  This allows us to have several jobs defined in the project and then opt to run a
+  specific one via a configuration change and redeploy.
+ */
+class MongoJobModule extends play.api.inject.Module with Logging {
+
+  override def bindings(environment: Environment, configuration: Configuration): collection.Seq[Binding[_]] = {
+    val workToDo = configuration.getOptional[Boolean]("mongoJob.enabled").getOrElse(false)
+
+    logger.info(s"Startup mongo job needs doing: $workToDo")
```
Comment (victorarbuesmallada) [inline app/uk/gov/hmrc/apihubapplications/mongojobs/MongoJobModule.scala:48]: do we want to log which mongo job is currently running (as in logging `mongoJob.className`)?

**Review by victorarbuesmallada — APPROVED** [review]
LGTM

File: app/uk/gov/hmrc/apihubapplications/mongojobs/HardDeleteMigrationJob.scala (hunk 1 above)
Comment (PaulCDurham) [reply app/uk/gov/hmrc/apihubapplications/mongojobs/HardDeleteMigrationJob.scala:110]: Yup, they are deleted applications. The code encountered an exception and failed to hard delete them.

**Pull request merged**